kong/cmd/init.lua

```diff
@@ -0,0 +1,88 @@
+local pl_app = require "pl.lapp"
+local log = require "kong.cmd.utils.log"
+
+local options = [[
+ --trace     with traceback
+ --v         verbose
+ --vv        debug
+]]
+
+local help = [[
+Usage: kong COMMAND [OPTIONS]
+
+The available commands are:
+ start
+ stop
+ reload
+ check
+ migrations
+ compile
+ version
+
+Options:
+]]..options
+
+local cmds = {
+  start = "start",
+  stop = "stop",
+  reload = "reload",
+  check = "check",
+  compile = "compile",
+  migrations = "migrations",
+  version = "version",
+  roar = "roar"
+}
+
+return function(args)
+  local cmd_name = table.remove(args, 1)
+  if not cmd_name then
+    pl_app(help)
+    pl_app.quit()
+  elseif not cmds[cmd_name] then
+    pl_app(help)
+    pl_app.quit("No such command: "..cmd_name)
+  end
+
+  local cmd = require("kong.cmd."..cmd_name)
+  local cmd_lapp = cmd.lapp
+  local cmd_exec = cmd.execute
+
+  if cmd_lapp then
+    cmd_lapp = cmd_lapp..options -- append universal options
+    args = pl_app(cmd_lapp)
+  end
+
+  -- check sub-commands
+  if cmd.sub_commands then
+    local sub_cmd = table.remove(args, 1)
+    if not sub_cmd then
+      pl_app.quit()
+    elseif not cmd.sub_commands[sub_cmd] then
+      pl_app.quit("No such command for "..cmd_name..": "..sub_cmd)
+    else
+      args.command = sub_cmd
```

Comment:
the long version: http://williamdurand.fr/2013/06/03/object-calisthenics/#dont-use-the-else-keyword

the short version: it's safer against future people refactoring and missing this logic branch to define a default and then only `if` the exception